demo/cpp/run_resnet.cc

```diff
@@ -15,70 +15,69 @@
 #include "dmlc/logging.h"
 #include "npy.hpp"
 
-struct NDArray {
-  std::vector<float> data;
-  std::vector<int64_t> shape;
-  int64_t ndim;
-  int64_t size;
-};
-
 bool is_big_endian();
-std::vector<std::vector<float>> RunInference(DLRModelHandle model,
-                                             const char* data_path,
-                                             const std::string& input_name);
+template <typename Scalar>
```

Comment:
Can we either rename Scalar to ScalarType or just T?

Reply:
Renamed to `T`